sklearn/metrics/tests/test_pairwise.py

```diff
@@ -893,3 +896,40 @@ def test_check_preserve_type():
                                                    XB.astype(np.float))
     assert_equal(XA_checked.dtype, np.float)
     assert_equal(XB_checked.dtype, np.float)
+
+
+@pytest.mark.parametrize("n_jobs", [1, 2, -1])
+@pytest.mark.parametrize("metric", ["seuclidean", "mahalanobis"])
+@pytest.mark.parametrize("dist_function",
+                         [pairwise_distances, pairwise_distances_chunked])
+@pytest.mark.parametrize("y_is_x", [True, False], ids=["Y is X", "Y is not X"])
+def test_pairwise_distances_data_derived_params(n_jobs, metric, dist_function,
+                                                y_is_x):
+    # check that pairwise_distances give the same result in sequential and
+    # parallel, when metric has data-derived parameters.
+    with config_context(working_memory=0.1):  # to have more than 1 chunk
+        rng = np.random.RandomState(0)
+
+        X = rng.random_sample((1000, 10))
+
+        if y_is_x:
+            Y = X
+            expected_dist_default_params = squareform(pdist(X, metric=metric))
+            if metric == "seuclidean":
+                params = {'V': np.var(X, axis=0, ddof=1)}
+            else:
+                params = {'VI': np.linalg.inv(np.cov(X.T)).T}
+        else:
+            Y = rng.random_sample((1000, 10))
+            expected_dist_default_params = cdist(X, Y, metric=metric)
+            if metric == "seuclidean":
+                params = {'V': np.var(np.vstack([X, Y]), axis=0, ddof=1)}
+            else:
+                params = {'VI': np.linalg.inv(np.cov(np.vstack([X, Y]).T)).T}
+
+        expected_dist_explicit_params = cdist(X, Y, metric=metric, **params)
```

Comment:
I don't like such a scipy test, but won't vote -1.

Reply:
What's wrong with that ?
You'd prefer to just test that `pairwise_distances(X, n_jobs=1) == pairwise_distances(X, n_jobs=2)` ?

Reply:
I guess n_jobs=2 is similar to n_jobs=-1? Just want to avoid some unnecessary tests but l'm fine with current version.

Reply:
Ahh, so you remove n_jobs=-1,that's also fine from my side :)

Reply:
Yes you were right, 2 and -1 are redundant.

Reply:
Apologies for the above comments.
I don't think we need to compare n_jobs=1 and n_jobs=2 since we've compared with scipy.

Reply:
It's hard to review on bed, apologies again for the meaningless comments above :)